crates/sui-faucet/src/main.rs

```diff
@@ -265,6 +266,55 @@ async fn request_gas(
     }
 }
 
+/// This function takes the old gas API and has a response that matches the new batch API interface.
+async fn request_gas_v1_interface(
+    Extension(state): Extension<Arc<AppState>>,
+    Json(payload): Json<FaucetRequest>,
+) -> impl IntoResponse {
+    // ID for traceability
+    let id = Uuid::new_v4();
+    info!(uuid = ?id, "Got new gas request.");
+    let result = match payload {
+        FaucetRequest::FixedAmountRequest(requests) => {
+            // We spawn a tokio task for this such that connection drop will not interrupt
+            // it and impact the recycling of coins
+            spawn_monitored_task!(async move {
+                state
+                    .faucet
+                    .send(
+                        id,
+                        requests.recipient,
+                        &vec![state.config.amount; state.config.num_coins],
+                    )
+                    .await
+            })
+            .await
+            .unwrap()
+        }
+        _ => {
+            return (
+                StatusCode::BAD_REQUEST,
+                Json(BatchFaucetResponse::from(FaucetError::Internal(
+                    "Input Error.".to_string(),
+                ))),
+            )
+        }
+    };
+    match result {
+        Ok(_) => {
+            info!(uuid =?id, "Request is successfully served");
+            (StatusCode::CREATED, Json(BatchFaucetResponse::from(id)))
```

Comment:
The response structure should be exactly the same as for `batch_request_gas`, i.e. it says that the request has been accepted, when in fact it has already successfully served the response.